src/rotues/QuizPage.jsx

```diff
@@ -1,34 +1,50 @@
+import { useEffect, useState } from "react";
 import { useQuiz } from "../context/quiz-context";
+import { useNavigate } from "react-router-dom";
 
 const QuizPage = () => {
-  const { quiz } = useQuiz();
+  const navigate = useNavigate();
+  const { quiz, quesCounter, setQuesCounter, score, setUserAnswers } =
+    useQuiz();
+
+  const [question, setQuestion] = useState(quiz.questions[quesCounter]);
+  useEffect(() => {
+    if (quesCounter >= quiz.numberOfQuestions) {
+      navigate("/result");
+    } else {
+      setQuestion(quiz.questions[quesCounter]);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
why this comment?

Reply:
So this comment is for disabling the warning/error that comes from eslint package related to the parameters inside the useEffect dependency array.